mne/time_frequency/spectrum.py

```diff
@@ -92,29 +93,10 @@ def plot_psd(self, fmin=0, fmax=np.inf, tmin=None, tmax=None, picks=None,
         -----
         %(notes_plot_psd_meth)s
         """
-        from ..io import BaseRaw
-
-        # triage reject_by_annotation
-        rba = dict()
-        if isinstance(self, BaseRaw):
-            rba = dict(reject_by_annotation=reject_by_annotation)
-
-        spectrum = self.compute_psd(
-            method=method, fmin=fmin, fmax=fmax, tmin=tmin, tmax=tmax,
-            picks=picks, proj=proj, n_jobs=n_jobs, verbose=verbose, **rba,
-            **method_kw)
-
-        # translate kwargs
-        amplitude = 'auto' if estimate == 'auto' else (estimate == 'amplitude')
-        ci = 'sd' if area_mode == 'std' else area_mode
-        # ↓ here picks="all" because we've already restricted the `info` to
-        # ↓ have only `picks` channels
-        fig = spectrum.plot(
-            picks='all', average=average, dB=dB, amplitude=amplitude,
-            xscale=xscale, ci=ci, ci_alpha=area_alpha, color=color,
-            alpha=line_alpha, spatial_colors=spatial_colors, sphere=sphere,
-            exclude=exclude, axes=ax, show=show)
-        return fig
+        frame = currentframe()
+        arginfo = getargvalues(frame)
+        init_kwargs, plot_kwargs = _triage_old_psd_kwargs(arginfo=arginfo)
+        return self.compute_psd(**init_kwargs).plot(**plot_kwargs)
```

Comment:
This shows up three times in this PR I think. Worth DRYing? You could probably use a private function plus frame.f_back to get there. But I could be wrong

Reply:
`frame.f_back` was the clue I needed. I had wanted to avoid that repetition but somehow didn't find that in my searches. Thanks.